src/main/webapp/app/shared/user-settings/ide-preferences/ide-settings.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit } from '@angular/core';
+import { ProgrammingLanguage } from 'app/entities/programming-exercise.model';
+import { Ide, ideEqual } from 'app/shared/user-settings/ide-preferences/ide.model';
+import { faPlus, faTrash } from '@fortawesome/free-solid-svg-icons';
+import { IdeSettingsService, PREDEFINED_IDE } from 'app/shared/user-settings/ide-preferences/ide-settings.service';
+
+@Component({
+    selector: 'jhi-ide-preferences',
+    templateUrl: './ide-settings.component.html',
+    styleUrls: ['../user-settings.scss'],
+})
```

Comment:
According to new Artemis guidelines, we should try to build standalone components whenever possible

Reply:
Please do this!